src/_igraph/graphobject.c

```diff
@@ -4633,24 +4633,24 @@ PyObject *igraphmodule_Graph_decompose(igraphmodule_GraphObject * self,
   if (igraphmodule_PyObject_to_connectedness_t(mode_o, &mode))
     return NULL;
 
-  igraph_vector_ptr_init(&components, 3);
+  igraph_graph_list_init(&components, 3);
```

Comment:
The return value should be checked here, and you should not initialize the graph list with a length of 3. It was okay for `igraph_vector_ptr_t` because that one would have only created null pointers, but now you are actually creating 3 graphs.

Reply:
Please double check the new version here